### PR TITLE
Implementation for Approle

### PR DESCRIPTION
**Pull request by uLan08**
Assigned `auth` to a variable since it is common between `tokenResponse` and newly added `approleResponse`. Added a basic example for approle as well.

### CODE REVIEW

File: example/auth_approle.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// file: example/approle.js
```
Comment (kr1sp1n) [inline example/auth_approle.js]: Please change to `example/auth_approle.js`.

Comment (uLan08) [inline example/auth_approle.js]: Done :smile:

**Comment by uLan08** [comment]
Thanks for the merge! Do you think you can publish a release anytime soon?

**Comment by kr1sp1n** [comment]
Released v0.5.9 right now 😉

**Comment by uLan08** [comment]
Great! Much appreciated!

**Pull request merged**